File: cpp/50a_1858A_Buttons.cpp
#include<bits/stdc++.h>

std::string solve(int a, int b, int c){
    if (c % 2 == 0){
        if (a > b){
            return "First";
        }
        else{
            return "Second";
        }
    }
    else{
        if(b > a){
            return "Second";
        }
        else{
            return "First";
        }
    }
}

int main(){
    int t, a, b, c;
    std::vector<std::string> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>a>>b>>c;
        ans.push_back(solve(a, b, c));
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/50a_1858A_Buttons.py
def solve(a, b, c):
    if c % 2 != 0:
        if b <= a:
            return "First"
        else:
            return "Second"
    else:
        if a <= b:
            return "Second"
        else:
            return "First"

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        a, b, c = input().split()
        a, b, c = int(a), int(b), int(c)
        ans.append(solve(a, b, c))
    for i in ans:
        print(i)

if __name__ == '__main__':
    main()